scheduler/server/persist_settings.go

```diff
@@ -0,0 +1,47 @@
+package server
+
+import (
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// Persistor interface for persisting scheduler settings and initializing the scheduler
+// from its persisted settings
+type Persistor interface {
+	PersistSettings(s Scheduler)
+	LoadSettings(s Scheduler)
+}
+
+// PersistedSettings the persisted scheduler settings structure for encoding/decoding as json
+type PersistedSettings struct {
+	ClassLoadPercents               map[string]int32  `json:"classLoadPercents"`
+	RequestorToClassMap             map[string]string `json:"requestorToClassMap"`
+	RebalanceMinimumDurationMinutes int               `json:"rebalanceMinimumDurationMinutes"`
+	RebalanceThreshold              int               `json:"rebalanceThreshold"`
+	Throttle                        int               `json:"throttle"`
+}
+
+// nopPersistor provides nop implementations of persist and load functions
+type nopPersistor struct{}
+
+func (p *nopPersistor) PersistSettings(s Scheduler) {
+	return
+}
+
+func (p *nopPersistor) LoadSettings(s Scheduler) {
+	return
+}
+
+func (s *statefulScheduler) LoadPersistedSettings(settings PersistedSettings) {
```

Comment:
If we have a nopPersistor do we even need this public API? What if we just always loaded persistent settings when the scheduler starts (with a default of using a nop unless a different one was provided)?

Reply:
reorged the code only the Persistor's methods are public, statefulScheduler's load/persist are not visible.